src/spaced_repetition/sm5.cljc

```diff
@@ -0,0 +1,147 @@
+(ns spaced-repetition.sm5)
+
+;; controls the rate at which EF is increased or decreased; must be a
+;; number between 0 and 1 (the greater it is the faster the changes of
+;; the OF matrix)."
+(def learn-fraction 0.5)
+(def min-ef 1.3)
+(def default-ef 2.5)
+
+(defn interpolate-with-ef [ef low high]
+  (let [t (/ (- ef min-ef) (- default-ef min-ef))]
+    (+ (* high t) (* low (- 1 t)))))
+
+(defn initial-optimal-factor [n ef]
+  (cond
+    (= 1 n) (interpolate-with-ef ef 5 8)
+    (= 2 n) (interpolate-with-ef ef 13 21)
+    :else (- ef 0.1)))
+
+(defn propagate [{[src-ef src-of] :last-set}
+                 dest-n dest-ef]
```

Comment:
I "propagate" the latest value set in any n>2 to every unset element of the table, instead of doing the filling algorithm proposed in the supermemo page.